src/structs/chain.rs

```diff
@@ -345,3 +345,5 @@ impl PartialEq for Chain {
         self.id() == other.id() && self.residues == other.residues
     }
 }
+
+impl Eq for Chain {}
```

Comment:
As Chain contains multiple f64 values it cannot implement Eq as f64 does not implement Eq.